File: src/client/java/com/somefrills/events/InventoryUpdateEvent.java
package com.somefrills.events;

import net.minecraft.network.packet.s2c.play.ScreenHandlerSlotUpdateS2CPacket;
import net.minecraft.item.ItemStack;

public class InventoryUpdateEvent {
    public ScreenHandlerSlotUpdateS2CPacket packet;
    public int slotId;
    public ItemStack stack;

    public InventoryUpdateEvent(ScreenHandlerSlotUpdateS2CPacket packet, ItemStack stack, int slotId) {
        this.packet = packet;
        this.stack = stack;
        this.slotId = slotId;
    }
}

File: src/main/kotlin/com/somefrills/events/InventoryUpdateEvent.kt
package com.somefrills.events

import net.minecraft.network.protocol.game.ClientboundContainerSetSlotPacket
import net.minecraft.world.item.ItemStack

class InventoryUpdateEvent(
    val packet: ClientboundContainerSetSlotPacket,
    @JvmField val stack: ItemStack,
    @JvmField val slotId: Int
) : FrillsEvent()